src/pseudopeople/interface.py

```diff
@@ -469,7 +469,9 @@ def validate_data_path_suffix(data_paths) -> None:
     return None
 
 
-def get_dataset_filepaths(source: Path, dataset_name: str) -> List[Path]:
+def get_dataset_filepaths(source: Union[Path, str], dataset_name: str) -> List[Path]:
```

Comment:
You don't actually need to do `if type(source) == str`. You can just Path(source) and even if it's already a Path it'll work.

Also, I think this is the wrong place to do this. If you follow the trail up (and the type hints), I think you should convert to Path at the start of _generate_dataset at the top of this script. In that way from there on out it's always a Path object and the type hints are all correct (including in the fetch_filepaths function which is where this get_dataset_filepaths function gets called)

Reply:
I was debating which one of these was the better way to do it.